### ISSUE TITLE
[Improvement proposal] Make autoequip all items on pickup optional

### ISSUE THREAD

**Issue by Chance4us** [diasurgical/devilutionX#1118]
Make autoequip all items on pickup optional because it is not always good if the player is equipped with a protective shield.
If a shield is being equipped automatically, it can happen that the character permanently blocks and can no longer move.
So it can lead to death by blocking.

**Comment by AJenbo** [comment]
How about simply not auto equipping shields?

**Comment by Chance4us** [comment]
Yes. I think it would be almost enough.
And one another exception when the items are almost broken.
So that they would not be destroyed by enemies.

**Comment by Chance4us** [comment]
Especially when I think of the "thinking cap", which has only one durability point.

**Comment by AJenbo** [comment]
I think making this optional might simply be the way to go then.

**Comment by Chance4us** [comment]
Agreed!

**Comment by Chance4us** [comment]
One more very important limitation.
Do not autoequip the item if it would result in death.
This could happen when a bad item reduces the health.

**Comment by galaxyhaxz** [comment]
Could also have it so you can't wear unidentified magic/unique items, so in practice it would only equip white/mundane items.

**Comment by AJenbo** [comment]
> Do not autoequip the item if it would result in death.
> This could happen when a bad item reduces the health.

The original didn't care about this with weapons.

**Comment by julealgon** [comment]
> Especially when I think of the "thinking cap", which has only one durability point.

How common is it for one to grab a Thinking Cap, in the dungeon, while not having anything equipped in the head slot @Chance4us ?

Some of the cases you are describing seem like very corner cases. I'd be hesitant in creating complex rules like that for situations that are never actually experienced in the game.

I think it would be best to support dynamically selecting which slot is "locked" and this information saved to the character. However that would be quite complex to implement today.

For now, I'll add a setting to ini that allows you to select which item types can be auto equipped.

In the future, this configuration should be available on a per-class basis, and then be able to be modified on a per-character basis.

We don't have a strong enough configuration system to support all that ATM.

**Comment by AJenbo** [comment]
> Some of the cases you are describing seem like very corner cases. I'd be hesitant in creating complex rules like that for situations that are never actually experienced in the game.

I agree but I think it expresses some general anxiety that some players probably will have, and so best to have it opt-in so that they feel that they committed to something and not that it's introduced without them knowing about how it may affect their play style.

I don't think you need to make it highly configurable, we can try and make the system intelligent to some extent, but for configurability I think on/off is all that is needed, most players that won't like this change probably won't want to tweak it either.

**Comment by AJenbo** [comment]
Some thing like:
```
[Game]
Auto Equip Items=0
```

**Comment by julealgon** [comment]
So you are proposing toggling the whole thing then, even including weapon auto-equip which existed in vanilla?

I guess that would be sensible (and much easier to do).

**Comment by AJenbo** [comment]
> So you are proposing toggling the whole thing then, even including weapon auto-equip which existed in vanilla?

Well, everything that wasn't in vanilla, so almost as easy :)

**Comment by Chance4us** [comment]
@julealgon It's really perfectly configurable now.
